afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java

```diff
@@ -181,19 +182,22 @@ private boolean nodeNameAlreadyExists(String name) {
         return childNodes.stream().filter(nodeInfo -> !nodeInfo.getId().equals(getId())).anyMatch(nodeInfo -> nodeInfo.getName().equals(name));
     }
 
-    public void archive(Path dir, boolean useZip) {
+    public void archive(Path dir, boolean useZip, boolean archiveDependencies) {
 
         Objects.requireNonNull(dir);
+        Path zipPath = null;
+        if (useZip) {
+            zipPath = dir.getParent().resolve(dir.getFileName() + ".zip");
```

Comment:
Should the zipFile name be based on the directory name or based on the current node ID?

Reply:
the directory name because when you archive you want an archive with the name you entered on the GUI

Reply:
I don't understand what the name given by the user: a filename or a directory name? If the username give a filename, why this filename is not pass to this method? If it's a dirname, it's an implementation choice done on the client side, not on the server side.

What do you think?

Reply:
The argument "dir" already contains the path where the directory must be created and the name of the directory given by the user.
When you archive, a directory with the name entered on the GUI is created and this function store all files to archive.
After that, we want zip this directory. So, this fonction create a zip with the same name that the directory previously created. And next we delete the directory.
Finally, we obtained a zip with all file to archived on and with the name entered by the user.

Reply:
I don't understand why you don't create a zip file directly then. It's not really efficient to use temporary files. The Java API offer everything you need...

Reply:
It would imply a complete rewrite of AppStorageArchive, also this issue is not related to this PR per se